subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/ApiGroovyCompiler.java

```diff
@@ -191,6 +162,10 @@ public boolean apply(File file) {
             CompositeStoppable.stoppable(classPathLoader, astTransformClassLoader).stop();
         }
 
+        if (specExecutingJavaCompiler.getResult() instanceof CompilationWithAnnotationProcessingResult) {
+            AnnotationProcessingResult annotationProcessingResult = ((CompilationWithAnnotationProcessingResult) specExecutingJavaCompiler.getResult()).getAnnotationProcessingResult();
```

Comment:
Why unpack and repack it?

Reply:
Good question. I wanted to make it more clear, that `CompilationWithAnnotationProcessingResult` in this case is the result of `ApiGroovyCompiler`. But let's continue the discussion [above](https://github.com/gradle/gradle/pull/7646#discussion_r232959474).